### DISCUSSION TITLE
Already granted / allow permission but camera still disable

### DISCUSSION THREAD

**Discussion by muksidin** [Budibase/budibase#11592] [Help] [answered]
Hello, i try to use barcode component and than in my browser ( i have tried Chrome and FireFox and already give grant and allow permission for camera), but the browser still can not access camera. I use the latest stable budibase version.
![Screenshot_20230824_202705_Chrome](https://github.com/Budibase/budibase/assets/5665115/0e7f402a-93f4-47da-9826-60f6abaef62c)

**Comment by ConorWebb96** [comment]
Hey @muksidin,

I've just tried this component on the latest version and everything seems to be working fine. 
Are you getting any errors within your console?

Have you tested your camera on other services to see if you can use it elsewhere? The `Your camera is disabled` message only shows up when your camera is disabled.

**Reply by muksidin** [reply]
yes i tested it in some devices, but result is same. Still `your camera is disable`.

**Reply by ConorWebb96** [reply]
Sorry for the confusion. I meant have you tried using your camera with other applications that aren't Budibase? E.g web cam tests etc.

**Reply by muksidin** [reply]
this is my console, i think this is my problem. I use budibase as self hosted in my LAN. any solutions?
![camera access](https://github.com/Budibase/budibase/assets/5665115/e4de675c-dd0c-46db-9a10-e15e797f889d)

**Reply by ConorWebb96** [reply]
Hey @muksidin,

What version of Budibase are you on?
What form of self-host are you using?
Are you able to use your camera on anything else other than Budibase?

**Reply by ConorWebb96** [reply]
Additionally @muksidin,

Can you look into doing this; https://www.chromium.org/Home/chromium-security/deprecating-powerful-features-on-insecure-origins/ you likely are using an unconventional install of Budibase that isn't hosted with Https or localhost.

Specifically you will need to change the setting; `chrome://flags/#unsafely-treat-insecure-origin-as-secure`

You also could look into changing your `platformUrl` to include the IP of the machine. As It might cause a discrepancy.

**Reply by muksidin** [reply]
*Version Budibase 2.9.18
*Form which generate automatically from table, i still understand what is exactly you mean in this question.
*Yes, other Budibase i already able it.

I try using FireFox but the result is same.

**Reply by muksidin** [reply]
Can you tell me more about changing `platformUrl`? in this case, i install budibase as self hosted in my server. and sure i access budibase using server's ip.

**Reply by ConorWebb96** [reply, marked as answer]
This issue is related to your specific hosting setup and how the browser is applying its security policies. if this is a dev server you can choose to override the browser security policies on a case by case basis. If this is intended for production i'd recommend fixing the infrastructure to make it work on https so that all browsers are happy.

A similar policy applies to Firefox which is why you're seeing the same issue there, I'm pretty sure most browsers will have a policy like this. You'll need to change this setting if you aren't running Budibase over localhost or https.

Google; https://www.chromium.org/Home/chromium-security/deprecating-powerful-features-on-insecure-origins/ 
Firefox; https://blog.mozilla.org/webrtc/camera-microphone-require-https-in-firefox-68/

You have a few different options, change the policies on the browsers to allow for http. Or modify your current self-host instance so it works over https/localhost instead of http.

The tool tip for the platformUrl; `Update the Platform URL to match your Budibase web URL. This keeps email templates and authentication configs up to date` It might not be relevant but could be something to look into. Its an environment variable.

You can find were to set the platformUrl within your organisation tab within Budibase.
<img width="644" alt="Screenshot 2023-08-25 at 12 36 34" src="https://github.com/Budibase/budibase/assets/126772285/23431768-430d-44a4-98c7-435829d0f46a">

**Reply by muksidin** [reply]
Thanks @ConorWebb96 , it works now.
